internal/cloudsql/refresh.go

```diff
@@ -49,15 +49,21 @@ func fetchMetadata(ctx context.Context, client *sqladmin.Service, inst connName)
 	defer func() { end(err) }()
 	db, err := client.Instances.Get(inst.project, inst.name).Context(ctx).Do()
 	if err != nil {
-		return metadata{}, fmt.Errorf("failed to get instance (%s): %w", inst, err)
+		return metadata{}, errtypes.NewRefreshError("failed to get instance metadata", inst.String(), err)
 	}
-
 	// validate the instance is supported for authenticated connections
 	if db.Region != inst.region {
-		return metadata{}, fmt.Errorf("provided region was mismatched - got %s, want %s", inst.region, db.Region)
+		return metadata{}, errtypes.NewRefreshError(
+			fmt.Sprintf("provided region was mismatched - got %s, want %s", inst.region, db.Region),
+			inst.String(),
+			nil,
+		)
```

Comment:
super-nit: I think it's generally cleaner to bias towards fewer lines over spreading the function call.
```suggestion
		msg := fmt.Sprintf("provided region was mismatched - got %s, want %s", inst.region, db.Region)
		return metadata{}, errtypes.NewRefreshError(msg, inst.String(), nil)
```